pkg/apis/app/v1alpha1/types.go

```diff
@@ -26,6 +28,22 @@ type Image struct {
 	Tag       string `json:"tag,omitempty"`
 }
 
+// IsEmpty verifies if this Image instance is empty
+func (i *Image) IsEmpty() bool {
+	return len(i.Namespace) == 0 &&
+		len(i.Domain) == 0 &&
+		len(i.Name) == 0 &&
+		len(i.Tag) == 0
```

Comment:
Maybe checking the Tag is too restricted? If not provided, tag should be "latest".

Reply:
We need this since any parameter is filled, the image would be a custom one.